src/compiler/crystal/macros/methods.cr

```diff
@@ -785,7 +785,7 @@ module Crystal
     def interpret(method : String, args : Array(ASTNode), named_args : Hash(String, ASTNode)?, block : Crystal::Block?, interpreter : Crystal::MacroInterpreter, name_loc : Location?)
       case method
       when "expressions"
-        interpret_check_args { ArrayLiteral.new(expressions) }
+        interpret_check_args { ArrayLiteral.map(expressions, &.itself) }
```

Comment:
Is there a reason to use `map` instead a `dup` of `expressions` (`ArrayLiteral.new(expressions.dup)`)?

Reply:
`ArrayLiteral.map` takes care of arguments like `Array(Arg)` with an `.as(ASTNode)` and this is how the rest of the interpreter does it.

Reply:
It is necessary here:

```crystal
module Crystal
  class Case
    def interpret(method : String, args : Array(ASTNode), named_args : Hash(String, ASTNode)?, block : Crystal::Block?, interpreter : Crystal::MacroInterpreter, name_loc : Location?)
      case method
      when "whens"
        interpret_check_args { ArrayLiteral.map whens, &.itself }
      # ...
      end
    end
  end
end
```

And I would prefer not having to remember which AST fields can use `ArrayLiteral.new(x.dup)` and which cannot, when both expressions mean the same thing.

Reply:
In this cases I think we should take the contribution as is, and later, if wanted, we ourselves can send a PR to change this. This is what the Ruby core team does.